sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -484,6 +488,59 @@ case class GpuCast(
     }
   }
 
+  private def castStructToString(input: GpuColumnVector): ColumnVector = {
+    var separatorColumn: ColumnVector = null
+    var spaceColumn: ColumnVector = null
+    val columns: ArrayBuffer[ColumnVector] = new ArrayBuffer[ColumnVector]()
+    val coreColumns: ArrayBuffer[ColumnVector] = new ArrayBuffer[ColumnVector]()
+
+    try {
+      withResource(GpuScalar.from(",", StringType)) { separatorScalar =>
+        separatorColumn = ColumnVector.fromScalar(separatorScalar, input.getRowCount().toInt)
+      }
+      withResource(GpuScalar.from(" ", StringType)) { separatorScalar =>
+        spaceColumn = ColumnVector.fromScalar(separatorScalar, input.getRowCount().toInt)
+      }
+      withResource(GpuScalar.from("[", StringType)) { bracketScalar =>
+        columns += ColumnVector.fromScalar(bracketScalar, input.getRowCount().toInt)
+      }
+      withResource(input.getBase().getChildColumnView(0)) { childView =>
+        columns += childView.asStrings()
+        coreColumns += childView.asStrings()
```

Comment:
Casting to strings twice is expensive. We should just increment the reference count instead.

Reply:
I was also leaking the second column, oops